### ISSUE TITLE
hicpp-use-equals-delete does not implement a HICPP guideline

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#65220]
The docs say it implements 12.5.1 from the guidelines, but that's not true. That rule is about Rule of 5 - ensure that your class has all 5 operations declared explicitly (and you can use = default or = delete if needed).

I can't find any similar rule in HICPP to "modernize-use-equals-delete" so I propose to remove the HICPP alias altogether.

**Comment by PiotrZSL** [comment]
I agree, this alias should be deleted. hicpp-special-member-functions implements 12.5.1 (not 12.5.7) and misc-unconventional-assign-operator actually implements 12.5.7, but no alias exists.

**Comment by PiotrZSL** [comment]
Personally I would remove this hicpp module, and all aliases and move all those checks into other modules like misc/bugprone/...
But maybe we could wait with this until some **real check aliases** support would be implemented.

**Comment by vbvictor** [comment]
Closing this since we no longer have hicpp module hence we can't really depend on their guidelines
